src/data/case.rs

```diff
@@ -138,7 +140,15 @@ impl Case {
     .text().await?;
 
         let case_information =
-            super::retrieve_escaped_json(&re::TRIAL_INFORMATION_REGEX, &case_script)?;
+            super::retrieve_escaped_json(&re::TRIAL_INFORMATION_REGEX, &case_script).map_err(
+                |x| {
+                    if x.root_cause().downcast_ref::<RegexNotMatched>().is_some() {
+                        anyhow!("The case with given ID {case_id} could not be found!")
```

Comment:
Not sure if this is an intentional choice here (feel free to ignore this comment in that case), but I think you could also use `x.context()` here:

```suggestion
                        x.context("The case with given ID {case_id} could not be found!")
```

Reply:
It was actually intentional here—I did not want to confuse users with an additional "Regex did not match" text in this case.